app/views/tag/_topicCard.html.erb

```diff
@@ -32,7 +34,7 @@
       </div>
 
       <div class="card-footer" style="background-color: inherit; border:none;">
-        <a style="padding-top:15px;text-decoration:underline;color:#808080;display:inline-block;" href="/tag/<%= tag.name %>"><%= Tag.counter(tag.name)-shown_nids.count - Tag.find_nodes_by_type(tag.name, type = 'note', limit = 3).where.not(nid: shown_nids).count %> <%= translation('tag.index.more_posts') %> &raquo;</a>
```

Comment:
Aha - so, i think here we actually do need to keep `-shown_nids.count` because that takes the total, but deducts the currently displayed posts because the link says `X more posts` -- does that make sense? I think your fix should still work because it resolves the spam filtering on line 119 of the controller. 

Can you re-add the offset for the already-displayed posts? Thank you!!!

Reply:
Just a doubt @jywarren 😅, for example, we have 2 posts with tag **one** and 1 post with tag **one** and **two** both,  
and all the 3 posts gets displayed in the topic card for tag **one**. So, topic card of **two** should show **0 more posts since the post is displayed already in tag-one**. I guess this is the thing we want here, right ?

Reply:
Aha - yes. So we could maybe use this syntax -- 

```ruby
[Tag.counter(tag.name)-shown_nids.count - Tag.find_nodes_by_type(tag.name, type = 'note', limit = 3).where.not(nid: shown_nids).count, 0].max
```

?